Dockerfile

```diff
@@ -1,16 +1,18 @@
-FROM golang:1.6
-
-RUN apt-get update && \
-    apt-get install -y apache2-utils && \
-    rm -rf /var/lib/apt/lists/*
+FROM golang:1.6-alpine
 
 ENV DISTRIBUTION_DIR /go/src/github.com/docker/distribution
 ENV DOCKER_BUILDTAGS include_oss include_gcs
 
 WORKDIR $DISTRIBUTION_DIR
 COPY . $DISTRIBUTION_DIR
 COPY cmd/registry/config-dev.yml /etc/docker/registry/config.yml
-RUN make PREFIX=/go clean binaries
+
+RUN set -ex \
+    && apk add --no-cache make git
+
+ENV CGO_ENABLED=0
+
+RUN make DOCKER_BUILDTAGS=netgo PREFIX=/go clean binaries
```

Comment:
It might be good to run `ldd` and confirm that `registry` is a static binary. I've sometimes had trouble with the Go linker's behavior changing between versions, or being influenced by code changes. To avoid creating an extra layer, this could be added to this `RUN` command with `&&`.